frontend/ui/widget/opdsbrowser.lua

```diff
@@ -474,6 +474,8 @@ function OPDSBrowser:downloadFile(item, format, remote_url)
     -- download to user selected directory or last opened dir
     local lastdir = G_reader_settings:readSetting("lastdir")
     local download_dir = G_reader_settings:readSetting("download_dir") or lastdir
+    item.author = item.author:gsub('[\\,%/,:,%*,%?,%",%<,%>,%|]','')
+    item.title = item.title:gsub('[\\,%/,:,%*,%?,%",%<,%>,%|]','')
```

Comment:
Just brainstorming here as it might be a bad idea, but there are characters that look similar or the same but aren't those characters so they could be replaced by something more specific.

Regular question mark: `?`
Fullwidth question mark: `？`
Small question mark: `﹖`
Ornamental question mark: `❓` (less likely to be supported)

Similarly you have `꞉` (Modifier Letter Colon), etc.

You can use the search on https://unicode-table.com/en/ to find many of these, but note that some (like `�`) have names like "replacement character" and the search isn't intelligent enough to show such characters as well.

Anyway, just a thought. I'm sure there must be plenty of objections to performing such replacements automagically.